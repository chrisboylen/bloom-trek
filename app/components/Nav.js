import React from 'react';
import { Image, StyleSheet } from 'react-native';
import { ButtonGroup } from 'react-native-elements';
import PropTypes from 'prop-types';

const Nav = ({ selectedIndex, updateIndex }) => {
  const flowers = () => <Image style={styles.icon} source={require('../../assets/icons/flower.png')} />;
  const home = () => <Image style={styles.icon} source={require('../../assets/icons/map.png')} />;
  const trails = () => <Image style={styles.icon} source={require('../../assets/icons/trails.png')} />;
  const buttons = [{ element: flowers }, { element: home }, { element: trails }];

  return (
    <ButtonGroup
      onPress={updateIndex}
      selectedIndex={selectedIndex}
      buttons={buttons}
      containerStyle={{
        height: '15%',
        borderWidth: 0,
        width: '100%',
        marginLeft: 0,
        marginTop: 0,
      }}
      innerBorderStyle={{ color: 'transparent' }}
      buttonStyle={{
        opacity: 0.5,
      }}
      selectedButtonStyle={{
        opacity: 1,
      }}
    />
  );
};

const styles = StyleSheet.create({
  icon: {
    height: 35,
    width: 35,
  },
});

Nav.propTypes = {
  selectedIndex: PropTypes.number,
  updateIndex: PropTypes.func,
};

export default Nav;
