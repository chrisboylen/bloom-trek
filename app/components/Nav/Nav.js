import React, { Component } from 'react';
import { View, Text, StyleSheet } from 'react-native';

export default class Nav extends Component {
  render() {
    return (
      <View>
        <Text style={styles.text}>Test</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  text: {
    justifyContent: 'center',
  },
});
