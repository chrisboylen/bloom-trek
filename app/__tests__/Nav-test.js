import React from 'react';
import { shallow } from 'enzyme';
import Nav from '../components/Nav';

describe('Nav', () => {
  let wrapper;
  let mockUpdateIndex;
  let selectedIndex;

  beforeEach(() => {
    mockUpdateIndex = jest.fn();
    selectedIndex = 0;
    wrapper = shallow(
      <Nav updateIndex={mockUpdateIndex} selectedIndex={selectedIndex} />,
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
