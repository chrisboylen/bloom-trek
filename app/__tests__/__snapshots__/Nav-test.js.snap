// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    selectedIndex={0}
    updateIndex={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "buttonStyle": Object {
        "opacity": 0.5,
      },
      "buttons": Array [
        Object {
          "element": [Function],
        },
        Object {
          "element": [Function],
        },
        Object {
          "element": [Function],
        },
      ],
      "containerStyle": Object {
        "borderWidth": 0,
        "height": "15%",
        "marginLeft": 0,
        "marginTop": 0,
        "width": "100%",
      },
      "innerBorderStyle": Object {
        "color": "transparent",
      },
      "onPress": [MockFunction],
      "selectedButtonStyle": Object {
        "opacity": 1,
      },
      "selectedIndex": 0,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "buttonStyle": Object {
          "opacity": 0.5,
        },
        "buttons": Array [
          Object {
            "element": [Function],
          },
          Object {
            "element": [Function],
          },
          Object {
            "element": [Function],
          },
        ],
        "containerStyle": Object {
          "borderWidth": 0,
          "height": "15%",
          "marginLeft": 0,
          "marginTop": 0,
          "width": "100%",
        },
        "innerBorderStyle": Object {
          "color": "transparent",
        },
        "onPress": [MockFunction],
        "selectedButtonStyle": Object {
          "opacity": 1,
        },
        "selectedIndex": 0,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
